core/src/plugins/stored/scsicrypto/scsicrypto-sd.cc

```diff
@@ -340,7 +340,7 @@ static bRC do_set_scsi_encryption_key(void* value)
      * Check if encryption key is needed for reading this volume.
      */
     P(crypto_operation_mutex);
-    if (!NeedScsiCryptoKey(dev->fd(), dev->dev_name, true)) {
+    if (!NeedScsiCryptoKey(dev->fd(), dev->archive_device, true)) {
```

Comment:
```suggestion
    if (!NeedScsiCryptoKey(dev->fd(), dev->archive_device_name, true)) {
```

Reply:
After a second thought I think that "archive_device" is the right name, as the directive that sets the value is "archive device",
See stored_conf.cc:

{"ArchiveDevice", CFG_TYPE_STRNAME, ITEM(res_dev, device_name), 0, CFG_ITEM_REQUIRED, NULL, NULL, NULL},

Reply:
Suggestion: archive_device_string